wasm-wrapper/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+#[macro_use]
+extern crate serde_derive;
+
+mod utils;
+
+use powersoftau::{cli_common::*, parameters::CeremonyParams, BatchedAccumulator};
+use snark_utils::{blank_hash, calculate_hash, UseCompression};
+
+use zexe_algebra::{Bls12_377, Bls12_381, PairingEngine as Engine, BW6_761};
+
+use wasm_bindgen::prelude::*;
+
+const COMPRESS_NEW_CHALLENGE: UseCompression = UseCompression::No;
+
+// When the `wee_alloc` feature is enabled, use `wee_alloc` as the global
+// allocator.
+#[cfg(feature = "wee_alloc")]
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[derive(Serialize)]
+pub struct ChallengeInfo {
+    challenge: Vec<u8>,
+    hash: Vec<u8>,
+}
+
+#[wasm_bindgen]
+pub struct Powersoftau {}
+
+#[wasm_bindgen(catch)]
+impl Powersoftau {
+    pub fn new(curve_kind: &str, batch_size: usize, power: usize) -> Result<JsValue, JsValue> {
+        let mut res = match curve_from_str(curve_kind).expect("123") {
+            CurveKind::Bls12_381 => new_challenge(&get_parameters::<Bls12_381>(batch_size, power)),
+            CurveKind::Bls12_377 => new_challenge(&get_parameters::<Bls12_377>(batch_size, power)),
+            CurveKind::BW6 => new_challenge(&get_parameters::<BW6_761>(batch_size, power)),
+        };
+        return Ok(JsValue::from_serde(&res.ok().unwrap()).unwrap());
+    }
+}
+
+pub fn get_parameters<E: Engine>(batch_size: usize, power: usize) -> CeremonyParams<E> {
+    CeremonyParams::<E>::new(power, batch_size)
+}
+
+pub fn new_challenge<T: Engine + Sync>(parameters: &CeremonyParams<T>) -> Result<ChallengeInfo, String> {
```

Comment:
We don't need `new_challenge` (nor `verify`) in WASM.
We only need `contribute` for both phases.